scripts/gitlab/build_and_test.sh

```diff
@@ -108,7 +108,7 @@ then
     $cmake_exe \
       -C ${hostconfig_path} \
       ${project_dir}
-    if ! $cmake_exe --build . -j; then
+    if ! $cmake_exe --build . -j 12; then
```

Comment:
I am not sure I understand this.  Can you elaborate?  Why "12"?

Reply:
First the main drawback: 12 is a value "optimized" for quartz.
"-j" maximize the number of threads to use. However, on quartz, we share a node for all builds. Throwing so many threads was counter productive, and could even have lead to shared memory issues. By reducing the maximum number of threads, I got the CI on quartz to run faster, and didn’t have to increase the allocation time.

Reply:
Quartz nodes have 2 sockets of 18 cores. Each cores supports 2 hyperthreads, and the 18 cores are organized by groups of 6 (sort of). Hence the 12 value. However, actually pining those threads to cores is again not optimal (per my tests). So I leave the thread orchestration to the OS.